### PR TITLE
Feature/variable context and apikey auth

### PR DESCRIPTION
**Pull request by brandon-michael-PFG**
I didn't see how to use the other pull request, so just opening a new one let me know if that's not what you intended.

A couple of things here, I wasnt sure what to use for the "short" version of header since h already applies for host, so I just used header, i.e. -header and --header as the short and long. 

Also, I was looking for the usage of BasicAuthentication as an Optional value from CLI to the client builder but didn't see where it was being used. Let me know if lines 51 thru 65 formsfeeder.client-cli/CommandLineClient isn't quite how you would handle getting those values on the client builder. Thank you!

### CODE REVIEW

File: formsfeeder.client-cli/src/main/java/formsfeeder/client/cli/CommandLineClient.java [hunk 1]
```diff
@@ -40,12 +42,29 @@ public static void mainline(String[] args, InputStream in, PrintStream out, Prin
...
-					  .build();	
+					  .plugin(cliParameters.plugin());
+
+			cliParameters.authParameters().ifPresent(
+					authParameters->{
+						if(authParameters instanceof BasicAuthParameters) {
+							ffClientBuilder.basicAuthentication(((BasicAuthParameters)authParameters).username(), ((BasicAuthParameters)authParameters).password());
+						}
+					}
+			);
```
Comment (rmcdouga) [inline formsfeeder.client-cli/src/main/java/formsfeeder/client/cli/CommandLineClient.java:57]: Thanks for adding this in (and adding a test)!  It was an oversight on my part, so much appreciated!

File: formsfeeder.client-cli/src/main/java/formsfeeder/client/cli/parameters/CommandLineAppParameters.java [hunk 2]
```diff
@@ -22,9 +23,13 @@
...
 public class CommandLineAppParameters implements AppParameters {
 	private static final String HOST_LOCATION_SHORT_OPTION = "h";
-	private static final String HOST_LOCATION_LONG_OPTION = "host";	
+	private static final String HOST_LOCATION_LONG_OPTION = "host";
+	private static final String CONTEXT_ROOT_SHORT_OPTION = "c";
+	private static final String CONTEXT_ROOT_LONG_OPTION = "contextroot";
 	private static final String USER_CREDENTIALS_SHORT_OPTION = "u";
-	private static final String USER_CREDENTIALS_LONG_OPTION = "user";	
+	private static final String USER_CREDENTIALS_LONG_OPTION = "user";
+	private static final String HEADER_SHORT_OPTION = "header";
```
Comment (rmcdouga) [inline formsfeeder.client-cli/src/main/java/formsfeeder/client/cli/parameters/CommandLineAppParameters.java:31]: So, everything looks great with just this one thing...  It's obviously a misnomer to call -header a "short" option.  Can we shorten it to "-hdr"?  That way it really is shorter but doesn't conflict with -h.  Sorry to be pedantic about it, but I think it's a small change and worth it to probably make it at this point.  If it was something internal only, I would accept it and change it later, but I'm worried that if we don't get this right, changing it later will seem arbitrary and disruptive.  I think we're better to get this right now before it comes gets entrenched by client code.

Comment (brandon-michael-PFG) [reply formsfeeder.client-cli/src/main/java/formsfeeder/client/cli/parameters/CommandLineAppParameters.java:31]: Yeah no problem at all, I guess I should have thought longer, have definitely used hdr before for things. Was stuck on "single letter" short options in my thinking though.

I was also thinking about context root short option, should that be cr maybe? would it ever make sense to add correlation id as an option in the cli and maybe that one makes more sense as c for sort option?

**Comment by rmcdouga** [comment]
Merged into the main branch.  Thanks for contributing!

**Pull request merged**